### ISSUE TITLE
Documentation for plugin event hooks

### ISSUE THREAD

**Issue by funwhilelost** [idno/known#2316]
Hey, guys.  The documentation is a little sparse on how the plugin event hooks are supposed to work.  I'm trying to debug an issue with the Akismet plug-in and not having much success.

I've got the plug-in set up on my Known site and I'm sifting through the logs, seeing that the plug-in is correctly identifying spam but it seems that `$event->setResponse(false)` doesn't have any effect on stopping the notation from being saved.  It's registering an event hook for the new annotation but it doesn't seem to be able to stop the annotation from being saved.  Any ideas on how this might be accomplished?

> registerEventHooks()
> If this exists, this function will be called when the plugin is loaded. It's suggested that you define your event hooks here.

**Comment by funwhilelost** [comment]
Okay, I haven't been able to make any progress on this and haven't heard anything over at `mapkyca/KnownAkismet` either.  My self-hosted Known is inundated with spam at this point so I'm going to find another solution.

**Comment by johanbove** [comment]
Hi @funwhilelost , is this issue still relevant? Are you still having spam problems?

**Comment by funwhilelost** [comment]
Yes, I'm still having rampant spam issues. I turned off public comments which has helped the volume but I still get trackback spam.

The Akismet plugin is logging the check, showing a clear spam flag, and the pertinent event handler is returning false. It's my impression that this should prevent the annotation from being created but it's not.

I would really love to get some more documentation about this as I went walking through the Core source code and couldn't place exactly how the event mechanism is supposed to work.

**Comment by johanbove** [comment]
Sorry to hear about the spam trouble. When you say "Akismet" plugin, you mean @mapkyca 's right? https://github.com/mapkyca/KnownAkismet 
Which version do you have? Perhaps this issue report should be moved to that repo?

**Comment by johanbove** [comment]
Just noticed that you already did just that. Sorry!
Here is the link: https://github.com/mapkyca/KnownAkismet/issues/3

**Comment by funwhilelost** [comment]
Yup! Just looking for more documentation on Known's plugin and event hooks
- there is almost nothing in the docs at last check.
